### ISSUE TITLE
Issue with `max-attack-time`: Attack Doesn't Stop After Specified Time

### ISSUE THREAD

**Issue by Kundan8000** [wapiti-scanner/wapiti#619]
I'm running the following command, but it isn't stopping the attack after 1 minute. Instead, it gets stuck, and I think I have to manually move to the next attack by pressing <kbd>CTRL</kbd> + <kbd>C</kbd> and selecting the next attack from the menu.
```shell
wapiti \
  --level 1 \
  -u http://testasp.vulnweb.com \
  -f json \
  -o report.json \
  -m all \
  --scope folder \
  --flush-session \
  -v2 \
  --no-bugreport \
  --max-attack-time 60
```

**Comment by Kundan8000** [comment]
@devl00p, please check. It gets stuck after a certain amount of time and either completes after 24 hours or sometimes takes up to 3 days to finish.

**Comment by devl00p** [comment]
Hello,

Which version are you using ?  Which module was running ?

**Comment by Kundan8000** [comment]
I'm running all modules as you can see in command & I've `Wapiti 3.2.0` installed on my server `Ubuntu 22.04.4 LTS`

**Comment by Kundan8000** [comment]
> Hello,
> 
> Which version are you using ? Which module was running ?

I'm running all modules as you can see in above command & I've `Wapiti 3.2.0` installed on my server `Ubuntu 22.04.4 LTS`

**Comment by Kundan8000** [comment]
Hi @devl00p, Have you found anything why it's getting stuck?

**Comment by Kundan8000** [comment]
Hi @devl00p, Thank you for responding. I’ve found another issue: the `brute_login_form` should automatically detect the login form and attempt brute-force login or SQL injection. Currently, this [site](https://www.demoblaze.com/) is vulnerable to **SQL injection** through the login form, which appears as a modal but wapiti fails to detect that.

**Comment by Kundan8000** [comment]
> The `--max-attack-time` is used to specify a max duration of processing for EACH attach module.
> 
> So if you specify 60 seconds with (for example) all 33 attack modules it will take approximately 30 minutes to execute. This is the expected behavior.
> 
> The CTRL+C behavior is not perfect though as the stop signal is checked after each crawled URL, not its mutations. Depending on the module it can takes quite some time but I'm looking to improve it thanks to asyncio cancellation features

But when I start the scan, all the modules take the specified time except for the `buster` module, which takes too long and gets stuck.

**Comment by devl00p** [comment]
This should be fixed by #626 (merged), which removes the old `stop_event` mechanism (checked only after each original request, so slow to react with many payloads/high latency) in favor of instantaneous asyncio task cancellation. That change should also apply to `--max-attack-time`, since `load_and_attack`'s handling was refactored in the same PR to properly collect `attacked_ids` even when the max time is reached mid-attack.

Closing — feel free to reopen if you can still reproduce the delay on a recent version.
